app/controllers/application_controller.rb

```diff
@@ -1,7 +1,25 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    @current_user ||= User.first
+  before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :set_ability
+
+  rescue_from CanCan::AccessDenied do |exception|
+    redirect_to root_url, alert: exception.message
   end
 
-  helper_method :current_user
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name email password password_confirmation bio role])
+    devise_parameter_sanitizer.permit(:account_update, keys: %i[name email password password_confirmation bio])
+
+    # Set default role to "user" if not provided during registration
+    return unless params[:action] == 'create' && params[:controller] == 'devise/registrations'
+
+    params[:user][:role] ||= 'user'
+  end
+
+  def set_ability
+    @ability = Ability.new(current_user)
+  end
```

Comment:
- [ ] Well done on your progress! In this milestone, while authorization rules weren't mandatory, you've taken the initiative to incorporate the `cancancan` gem. However, it seems that the configuration may need some adjustments. For a comprehensive understanding and proper utilization of `cancancan`, I recommend following the upcoming lesson that specifically covers this gem.

You have two options:

1. **Follow Documentation:** To get started, you can refer to the [cancancan GitHub repository](https://github.com/CanCanCommunity/cancancan?tab=readme-ov-file#installation) for detailed documentation and examples. This will guide you through the correct setup and usage of `cancancan`.

2. **Remove Existing Setting:** Alternatively, I recommend removing the existing settings since incorporating `cancancan` is part of your upcoming milestone. By doing so, you can streamline your codebase and avoid potential issues until you delve into the detailed setup during the next milestone.
![image](https://github.com/NoelLincoln/blog-app/assets/55840999/eea4893a-b0db-4f7b-a0c3-28f508c4a592)

If you ask for my opinion, I would recommend removing it for now, as it aligns with the project's structure and makes it easier to maintain. Great effort, and keep up the good work! 🌟🚀

Reply:
Thank you 👍